### ISSUE TITLE
Feature Request: Expose dynamic parameters for current route

### ISSUE THREAD

**Issue by osinga** [tightenco/ziggy#183]
### Expected Behavior
It can be useful to have all route parameters available as an object, e.g. to make asynchronous calls based on an ID. Taking an example from the README, a route like `route('events.venues.show', [1, 2])` would provide you with:

```javascript
{
    event: 1,
    venue: 2,
}
```

As for the API, I see two options:

**1. Based on the [`current()` implementation](https://github.com/tightenco/ziggy#current)**
```javascript
route().params()        // Returns `{ event: 1, venue: 2 }`
route().params('event') // Returns `1`
route().params('foo')   // Returns `undefined`
```

**2. Based on the [Vue Router implementation](https://router.vuejs.org/api/#route-object-properties)**
```javascript
route().params          // Returns `{ event: 1, venue: 2 }`
```

Both would fall back on an empty object.

### Current Behavior
None.

Is this something you'd be open to? If so, let me know your preferred implementation and I'd gladly create a PR.

**Comment by DanielCoulbourne** [comment]
I’d be open to this. It would need to be smart enough to handle arrays as well

**Comment by osinga** [comment]
For the query parameters, you mean? I was thinking of adding just the route parameters for now.
Vue Router, for example, has them split into `$route.params` and `$route.query`.

**Comment by DanielCoulbourne** [comment]
Right. Can you talk me through where the data would come from? I was assuming we would be parsing the `window.location` as we do for `route().current()`

If we were going to do that we would need to be smart enough to handle situations like this:

`http://example.com/posts?categories[]=blog&categories[]=news`
```javascript
route().params()          // returns { categories: ['blog', 'news'] }
```

**Comment by osinga** [comment]
Yeah, the data would indeed come from `window.location`. I've already created the [PR](https://github.com/tightenco/ziggy/pull/187), so you can see the implementation there.

I understand the need to handle arrays in query parameters, but my issue was initially aimed at route parameters. What I had in mind was the following:

```javascript
// http://example.com/users/1/posts?categories[]=blog&categories[]=news

route().params // Returns `{ user: '1' }`
route().query  // Returns `{ categories: ['blog', 'news'] }`
```

The first being my PR, and the second being something to be implemented separately. Personally, I think it makes more sense to set it up that way, so it's clear where it came from and to prevent name collision.
